src/main/java/org/mockito/BaseArgumentMatcher.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2017 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito;
+
+import org.mockito.internal.util.Decamelizer;
+
+/**
+ * Base class for classes that implement {@link ArgumentMatcher}.
+ *
+ * <p>This was provided to simplify migration from pre 2.x implementations; it provides a default
+ * implementation of {@link #toString()} that provides equivalent behavior to the pre 2.x version.
+ * It also resolves
+ */
+@Incubating
```

Comment:
Actually, I was planning on adding a version of this in a new 1.x version, say 1.10.20, so that developers can first switch to 1.10.20. Make changes (hopefully mostly automatically using error prone) so that their code will be compatible with 1.10.20 and 2.7.19 (or whatever version it goes in). They can then switch from 1.10.20 to 2.7.19 and if we have covered every incompatibility then everything will work as intended.

If you don't think this will be useful in the longer term then maybe its name should be changed to make that clearer, e.g. CompatArgumentMatcher or something. Maybe even in a separate package (separate build artifact) so developers who don't need it don't have to pay the cost for it.